metadata-ingestion/src/datahub/specific/custom_properties.py

```diff
@@ -0,0 +1,55 @@
+from typing import Generic, Iterable, Optional, TypeVar
+
+from datahub.emitter.mcp_patch_builder import MetadataPatchProposal
+from datahub.metadata.schema_classes import (
+    KafkaAuditHeaderClass,
+    MetadataChangeProposalClass,
+    SystemMetadataClass,
+)
+
+T = TypeVar("T", bound=MetadataPatchProposal)
+
+
+class CustomPropertiesPatchBuilder(Generic[T], MetadataPatchProposal):
```

Comment:
@hsheth2 : would like to get your input on this design. 
Salient points:

1. CustomPropertiesPatchBuilder adds patches to the parent builder instead of creating its own set of patches.
2. calls build() on the parent.
3. Offers a parent() method to get access to the parent builder if the caller needs it to chain further methods.

Reply:
This is different from how we implemented fieldPathPatchBuilder ... so would like to first get consensus on what good looks like, and then we can align both builders to the same pattern.